src/YoutubeAtom.tsx

```diff
@@ -274,25 +197,86 @@ export const YoutubeAtom = ({
         false,
     );
 
+    const player = useRef<YoutubePlayerType>();
+
     useEffect(() => {
-        if (!player) {
-            player = YouTubePlayer(`youtube-video-${videoMeta.assetId}`);
+        if (!player.current) {
+            player.current = YouTubePlayer(
+                `youtube-video-${videoMeta.assetId}`,
+            );
         }
-        const listener = player?.on(
+
+        let hasSentPlayEvent = false;
+        let hasSent25Event = false;
+        let hasSent50Event = false;
+        let hasSent75Event = false;
+
+        const listener = player.current?.on(
             'stateChange',
             (e: YT.PlayerEvent & YoutubeStateChangeEventType) => {
-                if (player) {
-                    onPlayerStateChangeAnalytics({
-                        e,
-                        setHasUserLaunchedPlay,
-                        eventEmitters,
-                        player,
-                    });
+                if (e.data === youtubePlayerState.PLAYING) {
+                    if (!hasSentPlayEvent) {
+                        eventEmitters.forEach((eventEmitter) =>
+                            eventEmitter('play'),
+                        );
+                        hasSentPlayEvent = true;
+
+                        setTimeout(() => {
+                            checkProgress();
+                        }, 3000);
+                    }
+
+                    const checkProgress = async () => {
+                        if (!player || !player.current) return null;
+                        const currentTime = await player.current?.getCurrentTime();
+                        const duration = await player.current?.getDuration();
+
+                        if (!duration || !currentTime) return;
+
+                        const percentPlayed = ((currentTime / duration) *
+                            100) as number;
+
+                        if (!hasSent25Event && 25 < percentPlayed) {
+                            eventEmitters.forEach((eventEmitter) =>
+                                eventEmitter('25' as VideoEventKey),
```

Comment:
Ditto on the coercion, I suspect we can just call the function with the `25` string?